cmd/podman/common/create.go

```diff
@@ -158,7 +158,7 @@ func GetCreateFlags(cf *ContainerCLIOpts) *pflag.FlagSet {
 	)
 	createFlags.StringArrayVarP(
 		&cf.env,
-		"env", "e", containerConfig.Env(),
```

Comment:
We want this.

Reply:
not sure it works as expected.  If I specify just `--env` then it takes the next flag as the argument.

Reply:
if we set the default here, it overrides the image's configuration

Reply:
I think we need an IsSet check on the flag, to only set it in SpecGen if it was explicitly set by the user

Reply:
> I think we need an IsSet check on the flag, to only set it in SpecGen if it was explicitly set by the user

I've not found a way to do it cleanly with string arrays/slices.  The Value is converted to a comma separated string, which doesn't work when there is a comma in the env variables.

Reply:
I think that's just String Slice - String Array is a proper array, and allows commas

Reply:
fixed